ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -136,6 +138,40 @@ public boolean setReplicaSealedState(ReplicaId replicaId, boolean isSealed) {
     }
   }
 
+  @Override
+  public boolean setReplicaStoppedState(List<ReplicaId> replicaIds, boolean isStopped) {
+    Set<String> replicasToUpdate = new HashSet<>();
+    for (ReplicaId replicaId : replicaIds) {
+      if (!(replicaId instanceof AmbryReplica)) {
+        throw new IllegalArgumentException(
+            "HelixParticipant only works with the AmbryReplica implementation of ReplicaId");
+      }
+      replicasToUpdate.add(replicaId.getPartitionId().toPathString());
+    }
+    boolean setResult;
+    synchronized (helixAdministrationLock) {
+      logger.trace("Getting stopped replicas from instanceConfig");
+      List<String> stoppedList = getStoppedReplicas();
+      if (stoppedList == null) {
+        stoppedList = new ArrayList<>();
+      }
+      Set<String> stoppedReplicas = new HashSet<>(stoppedList);
```

Comment:
let's rename these variables - the names are too close and confusing

Reply:
yes, they are a little confusing. I will change them to avoid misunderstanding.